src/main/java/ratpack/zipkin/ServerTracingModule.java

```diff
@@ -59,14 +59,21 @@ protected void configure() {
 
   @Provides
   public HttpTracing getTracing(final Config config, final ServerConfig serverConfig) {
-    Tracing tracing = Tracing.newBuilder()
+    Tracing.Builder builder = Tracing.newBuilder()
         .sampler(config.sampler)
         .currentTraceContext(new RatpackCurrentTraceContext())
         .localEndpoint(buildLocalEndpoint(config.serviceName, serverConfig.getPort(),
             serverConfig.getAddress()))
-        .localServiceName(config.serviceName)
-        .reporter(config.spanReporter)
-        .build();
+        .localServiceName(config.serviceName);
+
+    // for backwards comparability, prefer legacy reporter if defined otherwise use the
+    if (config.legacyReporter != null) {
```

Comment:
Ps i dont remember how I coded before but I think I probably coded to avoid a check like this. This check requires zipkin 1 in the runtime classpath (vs just compile). It isnt obvious until after you try to remove the dep ;) 300 KiB might not matter to people and easy enough to rewire later if asked.

Reply:
huh, so if we removed zipkin 1 from the runtime class path, do we get an exception or something?